### ISSUE TITLE
8.0 Cannot reconcile accounting entries of a partner linked to a res.users

### ISSUE THREAD

**Issue by alexis-via** [odoo/odoo#11931]
Here is the scenario to reproduce the bug on Odoo v8:
1) connect to runbot v8 as admin
2) Check that the Demo user is in the Accountant group (so that he can create accounting entries and can reconcile entries) but not in the "Administration > Access rights" group (so that he doesn't have write access on res.users)
3) Logout and login back as Demo user
4) Validate an expense note of 200 € for the Employee "Administrator" ; it creates an account move of 200 € from X2110 Expenses to X1111 Creditors with Partner = Administrator.
5) Create an account move to pay this expense note: in the bank journal, create a move of 200 € from X1111 Creditors to X11004 Bank account with Partner = Administrator.
6) In the Journal Items, filter on X1111 and Partner = Administrator. Select the 2 account move lines of 200 € and do "More > Reconcile Entries" ; Odoo should propose a full reconciliation without write-off: click on "Reconcile".
=> You get the following error:
```
Sorry, you are not allowed to modify this document. Only users with the following access level are currently allowed to do that:
- Administration/Access Rights
(Document model: res.users) 
```

Code analysis:
This is a regression that has been introduced by the fix of @odony  that introduced this change: https://github.com/odoo/odoo/commit/a7c2377500694f98de75b8a8ea8a699cef55a7a6
With this piece of code, when there is a write() on a res.partner which is linked to a res.users, then the system checks that you have write access on res.users.
But, in the above scenario, when you reconcile some account move lines linked to a partner which is a res.users, Odoo will do a write() on res.partner with vals = {'last_reconciliation_date': '2016-05-04 21:08:17'}... but the Accountant is not supposed to have write access on res.users ! So Odoo blocks the reconciliation...

**Comment by odony** [comment]
Thanks for the report, should be fixed in 7.0 as of 7cafe6a9c16c34ba0352f83f4f1b90d2623a2cda, forwarded in 8.0 via c8dc4c8daf8a3a2f8e52448c164a30d3cbeb524c, and soon to be in 9.0.

**Comment by alexis-via** [comment]
@odony Thanks for your fix !
